rerun_py/src/catalog/connection_handle.rs

```diff
@@ -269,48 +269,62 @@ impl ConnectionHandle {
                 .map_err(to_py_err)?
                 .into_inner();
 
-            // cheat: we know there is only one item in the stream
-            let tasks_statuses = if let Some(response) = response_stream.next().await {
-                response
+            let mut errors: Vec<String> = Vec::new();
+
+            // loop until all the tasks are done or the timeout is reached: both cases
+            // will complete the stream
+            while let Some(response) = response_stream.next().await {
+                let item = response
                     .map_err(to_py_err)?
                     .data
                     .ok_or_else(|| PyValueError::new_err("received response without data"))?
                     .decode()
-                    .map_err(to_py_err)?
-            } else {
-                return Err(PyConnectionError::new_err("no response from task"));
-            };
-
-            // TODO(andrea): this is a bit hideous. Maybe the idea of returning a dataframe rather
-            // than a nicely typed object should be revisited.
-
-            let exec_statuses = tasks_statuses
-                .column_by_name("exec_status")
-                .ok_or_else(|| {
-                    PyValueError::new_err("bug: exec_status column not found in response")
-                })?
-                .try_downcast_array_ref::<arrow::array::StringArray>()
-                .map_err(to_py_err)?;
-
-            let msgs = tasks_statuses
-                .column_by_name("msgs")
-                .ok_or_else(|| PyValueError::new_err("bug: msgs column not found in response"))?
-                .try_downcast_array_ref::<arrow::array::StringArray>()
-                .map_err(to_py_err)?;
-
-            // report the first non success error
-            for (i, status) in exec_statuses.iter().enumerate() {
-                let status = status.ok_or_else(|| PyValueError::new_err("missing status value"))?;
+                    .map_err(to_py_err)?;
 
-                if status != "success" {
-                    return Err(PyValueError::new_err(format!(
-                        "task failed: {}",
-                        msgs.value(i)
-                    )));
+                // TODO(andrea): all this column unrwapping is a bit hideous. Maybe the idea of returning a dataframe rather
+                // than a nicely typed object should be revisited.
+                let schema = item.schema();
+                let col_indices = ["task_id", "exec_status", "msgs"]
```

Comment:
pretty please extend `QueryTasksOnCompletionResponse` on the model of e.g. `CreatePartitionManifestsResponse`, with `const`s for field names and, ideally, a `schema()` method 🙏🏻